src/nvim/optionstr.c

```diff
@@ -1792,6 +1792,16 @@ static const char *did_set_statustabline_rulerformat(optset_T *args, bool rulerf
   }
   const char *errmsg = NULL;
   char *s = *varp;
+
+  // reset statusline to default when setting global option and empty string is being set
+  if (!rulerformat && !statuscolumn
+      && ((args->os_flags & OPT_GLOBAL) || !(args->os_flags & OPT_LOCAL))
+      && s[0] == NUL) {
+    xfree(*varp);
+    *varp = xstrdup(args->os_def->string.data);
+    s = *varp;
+  }
+
```

Comment:
Why is `os_def` needed? Can't you check if `os_idx` is `kOptStatusline` and use its `def_val` directly?

Reply:
That was actually my first idea.

`os_idx` is an index to `static` `options` array in `options.generated.h` . I assumed it was meant to be only imported in options.c . If I can include it in optionstr.c too then your solution is better.

Reply:
![image](https://github.com/user-attachments/assets/b58e1036-830e-46dd-942c-c849f8a6b349)

@zeertzjq  Considering it's a static global variable. Including the header to multiple .c files would lead to linker error right? Global being defined twice?

Isn't just passing the necessary info better in that case as I've done with os_def? It's a pointer anyway doesn't really have any overhead.

Reply:
You can add a function in option.c to retrieve `def_val`.

Reply:
There is `get_option_default()`

Reply:
> There is `get_option_default()`

Ooo didn't see that. Will update.